src/main/java/frc/trigon/robot/components/XboxController.java

```diff
@@ -70,12 +68,20 @@ public void setDeadband(double deadband) {
         this.deadband = deadband;
     }
 
-    public void rumble(double durationSeconds, double power) {
-        if (stopRumblingCommand != null)
-            stopRumblingCommand.cancel();
-        stopRumblingCommand = Commands.getDelayedCommand(durationSeconds, () -> getHID().setRumble(GenericHID.RumbleType.kBothRumble, 0));
-        stopRumblingCommand.schedule();
-        getHID().setRumble(GenericHID.RumbleType.kBothRumble, power);
+    public synchronized void rumble(double durationSeconds, double power) {
+        newRumbleCommandCalled = true;
+
+        new StartEndCommand(
+                () -> {
+                    getHID().setRumble(GenericHID.RumbleType.kBothRumble, power);
+                    newRumbleCommandCalled = false;
+                },
+                () -> {
+                    if(!newRumbleCommandCalled) {
+                        getHID().setRumble(GenericHID.RumbleType.kBothRumble, 0);
+                    }
+                }
```

Comment:
At this point the logic is too complicated and ugly. I would just revert to how it was before . Simpler and more readable.

Reply:
aight